File: 프로그래머스/programmers/etc/해시/의상.py
def solution(clothes):
    answer = 1
    cloth_dict = {}
    for v, k in clothes :
        if not k in cloth_dict :
            cloth_dict[k] = 0
        cloth_dict[k] += 1
    print(cloth_dict)

    for i in cloth_dict.values() :
        answer *= i + 1

    return answer - 1

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        HashMap<String, Integer> clothHash = new HashMap<>();

        // 배열 저장 (각 종류별 갯수 저장)
        for (int i=0; i<clothes.length; i++) {
            String clothType = clothes[i][1];
            String clothName = clothes[i][0];
            if (clothHash.containsKey(clothType)) clothHash.put(clothType, clothHash.get(clothType) + 1);
            else clothHash.put(clothType, 1);
        }

        // 조합
        int answer =1;
        for (int i: clothHash.values()) answer *= (i+1); 

        return answer-1; // 아무것도 안 입은 경우 제외
    }
    }
